cli/config/configfile/file_test.go

```diff
@@ -157,3 +188,139 @@ func TestGetAllCredentials(t *testing.T) {
 	expected["example.com/foo"] = exampleAuth
 	assert.Equal(t, expected, authConfigs)
 }
+
+func TestGetAllCredentialsCredsStore(t *testing.T) {
+	configFile := New("filename")
+	configFile.CredentialsStore = "test_creds_store"
+	testServerAddress := "example.com"
+	expectedAuth := types.AuthConfig{
+		Username: "user",
+		Password: "pass",
+	}
+
+	testCredsStore := NewMockNativeStore(map[string]types.AuthConfig{testServerAddress: expectedAuth})
+
+	newNativeStore = func(configFile *ConfigFile, helperSuffix string) credentials.Store {
+		return testCredsStore
+	}
+
+	authConfigs, err := configFile.GetAllCredentials()
+	require.NoError(t, err)
+
+	expected := make(map[string]types.AuthConfig)
+	expected[testServerAddress] = expectedAuth
+	assert.Equal(t, expected, authConfigs)
+	assert.Equal(t, 1, testCredsStore.(*mockNativeStore).GetAllCallCount)
+}
+
+func TestGetAllCredentialsCredHelper(t *testing.T) {
+	testFileStoreServerAddress := "example.com"
+	testCredHelperSuffix := "test_cred_helper"
+	testCredHelperServerAddress := "credhelper.com"
+
+	expectedFileStoreAuth := types.AuthConfig{
+		Username: "file_store_user",
+		Password: "file_store_pass",
+	}
+	expectedCredHelperAuth := types.AuthConfig{
+		Username: "cred_helper_user",
+		Password: "cred_helper_pass",
+	}
+
+	configFile := New("filename")
+	configFile.CredentialHelpers = map[string]string{testCredHelperServerAddress: testCredHelperSuffix}
+	configFile.AuthConfigs[testFileStoreServerAddress] = expectedFileStoreAuth
+
+	testCredHelper := NewMockNativeStore(map[string]types.AuthConfig{testCredHelperServerAddress: expectedCredHelperAuth})
+
+	newNativeStore = func(configFile *ConfigFile, helperSuffix string) credentials.Store {
+		return testCredHelper
+	}
+
+	authConfigs, err := configFile.GetAllCredentials()
+	require.NoError(t, err)
+
+	expected := make(map[string]types.AuthConfig)
+	expected[testFileStoreServerAddress] = expectedFileStoreAuth
+	expected[testCredHelperServerAddress] = expectedCredHelperAuth
+	assert.Equal(t, expected, authConfigs)
+	assert.Equal(t, 0, testCredHelper.(*mockNativeStore).GetAllCallCount)
+}
+
+func TestGetAllCredentialsCredStoreAndCredHelper(t *testing.T) {
```

Comment:
Is this testing anything that isn't already covered by `TestGetAllCredentialsCredHelperOverridesDefaultStore` ?

Reply:
It tests that values from both are returned, if they don't conflict, rather than just one of the other. `TestGetAllCredentialsCredHelperOverridesDefaultStore` tests for the conflicting case.